`make clean PAT=false` yields an error

Trying to run `make clean PAT=false` yields the following error:

```sh
$ odkrun make clean PAT=false
make[1]: *** No rule to make target 'pattern_clean'.  Stop.
make[1]: Leaving directory '/work/src/ontology'
make: *** [Makefile:1064: clean] Error 2
```

This is a mistake introduced when we switched to using Make conditionals (#1168): The `pattern_clean` target is defined _within_ the `ifeq ($(PAT),true)` block:

```Make
ifeq ($(PAT),true)

[...]
pattern_clean:
    rm -f $(PATTERN_CLEAN_FILES)

[...]

endif
```

so the entire rule does not exist under `PAT=false`, even though it is a prerequisite of the general `clean` rule.

This is not critical but very annoying, because while there is no reason to call `make clean PAT=false` in itself, it is common to explicitly call the `clean` target as part of a larger command, as in:

```sh
$ odkrun make clean uberon.owl PAT=false
```

to rebuild the ontology out of a clean state.

The fix is simply to move the `pattern_clean` rule out of the conditional block.

--- a/template/src/ontology/Makefile.jinja2
+++ b/template/src/ontology/Makefile.jinja2
@@ -804,21 +804,19 @@ endif
 # DOSDP Templates/Patterns
 # ----------------------------------------
 
-ifeq ($(PAT),true)
-
 ALL_PATTERN_FILES=$(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml)
 ALL_PATTERN_NAMES=$(strip $(patsubst %.yaml,%, $(notdir $(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml))))
 
 PATTERN_CLEAN_FILES=../patterns/all_pattern_terms.txt \
-	$(DOSDP_OWL_FILES_DEFAULT) $(DOSDP_TERM_FILES_DEFAULT) \{% if project.pattern_pipelines_group is defined %}
-	{% for pipeline in project.pattern_pipelines_group.products -%}
-		$(DOSDP_OWL_FILES_{{ pipeline.id.upper() }}) $(DOSDP_TERM_FILES_{{ pipeline.id.upper() }}) {% endfor %}{% endif %}
+	$(DOSDP_OWL_FILES_DEFAULT) $(DOSDP_TERM_FILES_DEFAULT){% if project.pattern_pipelines_group is defined  -%}
+	{% for pipeline in project.pattern_pipelines_group.products %} \
+	$(DOSDP_OWL_FILES_{{ pipeline.id.upper() }}) $(DOSDP_TERM_FILES_{{ pipeline.id.upper() }}){% endfor %}{% endif %}
 
-# Note to future generations: prepending ./ is a safety measure to ensure that 
-# the environment does not malicously set `PATTERN_CLEAN_FILES` to `\`.
 .PHONY: pattern_clean
 pattern_clean:
 	rm -f $(PATTERN_CLEAN_FILES)
+
+ifeq ($(PAT),true)
 
 .PHONY: patterns
 patterns dosdp:

